src/k2/configurator/Configurator.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+MIT License
+
+Copyright(c) 2020 Futurewei Cloud
+
+    Permission is hereby granted,
+    free of charge, to any person obtaining a copy of this software and associated documentation files(the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and / or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in all copies
+    or
+    substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS",
+    WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+    DAMAGES OR OTHER
+    LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+*/
+
+#include "Configurator.h"
+#include "ConfiguratorDTO.h"
+
+
+#include <k2/appbase/AppEssentials.h>
+#include <k2/appbase/Appbase.h>
+#include <k2/common/Common.h>
+#include <k2/logging/Log.h>
+#include <k2/common/Common.h>
+#include <k2/transport/PayloadSerialization.h>
+#include <k2/transport/Status.h>
+
+namespace k2 {
+
+namespace log {
+inline thread_local k2::logging::Logger configurator("k2::configurator_server");
+}
+
+Configurator::Configurator() {
+    K2LOG_I(log::configurator, "ctor");
+}
+
+Configurator::~Configurator() {
+    K2LOG_I(log::configurator, "dtor");
+}
+
+seastar::future<> Configurator::start() {
+    K2LOG_I(log::configurator, "Registering message handlers");
+
+    RPC().registerRPCObserver<SET_ConfiguratorRequest, SET_ConfiguratorResponse>(InternalVerbs::CONFGURATOR_SET, [this](SET_ConfiguratorRequest&& request) {
+                K2LOG_I(log::configurator, "Received set for key: {}", request.key);
+                auto key = request.key;
+                auto value = request.value;
+
```

Comment:
We probably need to validate the incoming value here before storing it in the config. you should check the boost  map API to see if it does any config-related validation. If not, we should do it ourselves and return the appropriate status back to the RPC caller.

For log-related config, we need to apply the new value(for both set/delete) to the log system. The logic for that is here: https://github.com/futurewei-cloud/chogori-platform/blob/master/src/k2/appbase/Appbase.cpp#L81 and we may have to refactor it to reuse it.

You can also avoid code duplication like so:
``` c++
return [request=std::move(request)] {
    auto applier = [request] (auto& config) {
        config.emplace(key, boost::program_options::variable_value(value, ""));
        boost::program_options::notify(config);
        return seastar::make_ready_future();
    };
    if (request.applyToAll) {
        return ConfigDist().invoke_on_all(applier);
    }
    return applier();
    }()
    .then([] {
        return RPCResponse(Statuses::S201_Created("ConfigSet accepted"), ConfigSetResponse{});
    });
```